Templates/CSharp/Base/SharedCSharp.template.tt

```diff
@@ -5,13 +5,27 @@
 CustomT4Host host   = (CustomT4Host)Host;
 OdcmModel model     = host.CurrentModel;
 var writer          = (CodeWriterCSharp)host.CodeWriter;
+bool logTemplateSrc = false;
+
 #>
 <#=writer.WriteHeader()#>
 
 // **NOTE** This file was generated by a tool and any changes will be overwritten.
 
+<# if (logTemplateSrc) { #>
+// Template Source: <#= TemplateName(host.TemplateFile) #>
+<# } #>
 <#+
 
+/// <summary>
```

Comment:
This logic is now being used in Android and .NET. Might be worth pulling this up into shared template writer code (extensions) to be re-used in both places.